server/src/server.ts

```diff
@@ -29,6 +29,8 @@ import { LogicManager } from '@accordproject/ergo-compiler';
 import { ModelFile } from '@accordproject/concerto-core';
 
 const util = require('util');
+const findUp = require('find-up');

```

Comment:
Can we do this without adding a new dependency?

Reply:
@dselman I have fixed the issue in my latest commit, please review. The problem was in the condition of the while loop, because of which the program was never exiting the loop for directories that were not having a valid package.json file.